cirq/_compat.py

```diff
@@ -251,12 +274,11 @@ class Wrapped(ModuleType):
         def __getattr__(self, name):
             if name in deprecated_attributes:
                 deadline, fix = deprecated_attributes[name]
-                warnings.warn(
+                _validate_deadline(deadline)
```

Comment:
Why is the deadline being validated at call time instead of at define time?

Reply:
oh, that is an oversight, thanks for catching that!